src/extension.ts

```diff
@@ -3,34 +3,54 @@
+import * as languageclient from 'vscode-languageclient';
+
 import { activateConfigurePipeline } from './configure/activate';
+import { extensionVariables } from './configure/model/models';
+import * as logger from './logger';
+import * as schemaassociationservice from './schema-association-service';
+import * as schemacontributor from './schema-contributor';
 
-const myExtensionId = 'azure-pipelines';
-const telemetryVersion = generateVersionString(vscode.extensions.getExtension(`ms-azure-devops.${myExtensionId}`));
-const telemetryKey = 'ae672644-d394-497c-8c57-98f6eac32342';
+let perfStats = {
+    loadStartTime: Date.now(),
+    loadEndTime: undefined
+};
 
-let reporter: TelemetryReporter;
+const configurePipelineEnabled: boolean = vscode.workspace.getConfiguration('[azure-pipelines]', null).get('configure') ? true : false;
 
 export async function activate(context: vscode.ExtensionContext) {
     logger.log('Extension has been activated!', 'ExtensionActivated');
+    setTelemetryReporter(context);
+
+    await callWithTelemetryAndErrorHandling('azurePipelines.activate', async (activateContext: IActionContext) => {
+        activateContext.telemetry.properties['configurePipelineEnabled'] = `${configurePipelineEnabled}`;
+        await activateYmlContributor(context, activateContext);
+        if (configurePipelineEnabled) {
+            await activateConfigurePipeline();
+        }
+    });
 
-    logger.log(`Spinning up telemetry client for id ${myExtensionId}, version ${telemetryVersion}`);
-    reporter = new TelemetryReporter(myExtensionId, telemetryVersion, telemetryKey);
-    context.subscriptions.push(reporter);
+    return schemacontributor.schemaContributor;
+}
+
+function setTelemetryReporter(context: vscode.ExtensionContext) {
+    // Register ui extension variables is required to be done for telemetry to start flowing for extension activation and other events.
+    // It also facilitates registering command and called events telemetry.
+    extensionVariables.reporter = createTelemetryReporter(context);
+    extensionVariables.outputChannel = vscode.window.createOutputChannel('Azure Pipelines');
+    context.subscriptions.push(extensionVariables.outputChannel);
+    extensionVariables.context = context;
+    extensionVariables.ui = new AzureUserInput(context.globalState);
+    registerUIExtensionVariables(extensionVariables);
+}
 
-    try {
-        reporter.sendTelemetryEvent('extension.activate');
-    } catch(e) {
-        // if something bad happens reporting telemetry, swallow it and move on
-        logger.log(e.toString());
+async function activateYmlContributor(context: vscode.ExtensionContext, activateContext: IActionContext) {
+    if (activateContext) {
+        activateContext.telemetry.properties.isActivationEvent = 'true';
+        activateContext.telemetry.measurements.mainFileLoad = (perfStats.loadEndTime - perfStats.loadStartTime) / 1000;
```

Comment:
loadEndTime will be undefined here

Reply:
this is initialized at the bottom of the file, globally, hence it gets defined as soon as the file is loaded in memory

Reply:
yes, you are correct. but even then the endtime is useless. It should be ticked when activate completes. Isn't that what you are trying to measure?